src/main/java/org/springframework/retry/annotation/RetryConfiguration.java

```diff
@@ -51,6 +52,8 @@
 import org.springframework.util.ReflectionUtils;
 import org.springframework.util.ReflectionUtils.MethodCallback;
 
+import static org.springframework.beans.factory.config.BeanDefinition.ROLE_INFRASTRUCTURE;
```

Comment:
We prefer to avoid `import static` in production code. 
While it might fix the immediate warning, I wonder if it is still legit to keep the logic this `InitializingBean` way...
May be better to revise it in favor of `SmartInitializingSingleton`, so there won't be dependencies on other beans during this `Advisor` introspection.
I don't tell that we need to do this right now since it might be slightly a breaking change even if this class is not intended to be used somehow in the target project.

Please, update the copyright header to he current year and add your name to the `@author` list.
Then we can merge this as is and revise the possibilities later on.
Thanks

Reply:
I changed what you requested.